internal/modifier/gated.go

```diff
@@ -37,8 +37,9 @@ import (
 //
 // If not devices are selected, no changes are made.
 func NewFeatureGatedModifier(logger logger.Interface, cfg *config.Config, image image.CUDA, driver *root.Driver, hookCreator discover.HookCreator) (oci.SpecModifier, error) {
-	if devices := image.VisibleDevicesFromEnvVar(); len(devices) == 0 {
-		logger.Infof("No modification required; no devices requested")
+	devices := image.VisibleDevices()
+	if len(devices) == 0 {
+		logger.Debugf("No modification required; no devices requested")
```

Comment:
This is a functional change and should at the very least be a separate commit from the refactoring which *should* only include non-functional changes.